### PR TITLE
Updated step to modify kube-dns deployment for fixed identity

### PR DESCRIPTION
**Pull request by ackerman80**
Updated step to modify kube-dns deployment for fixed identity
Updated to included dependencies for cert generation scripts

Signed-off-by: Arvind Soni <arvindsoni@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5518)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-docs-please

**Comment by ianvernon** [comment]
@ackerman80 if you add a preceding `/` before the test trigger phrase, it will not work.

**Review by ianvernon — APPROVED** [review]

**Review by aanm — CHANGES_REQUESTED** [review]
This might require more steps. Blocking before merging

**Comment by ackerman80** [comment]
@aanm agreed. let me know your edits and I will update the docs along with testing them. Particularly unsure why we were getting multiple dns replica-sets today.

**Comment by aanm** [comment]
@ackerman80 this will work:
for coredns
```
kubectl patch -n kube-system deployment/coredns --type merge -p '{"spec":{"template":{"metadata":{"labels":{"io.cilium.fixed-identity":"kube-dns"}}}}}'
```
for kube-dns
```
kubectl patch -n kube-system deployment/kube-dns --type merge -p '{"spec":{"template":{"metadata":{"labels":{"io.cilium.fixed-identity":"kube-dns"}}}}}'
```
you can also mention that by, default kubernetes keeps the old replicaset but it will not be running. If the deployment is deleted all replicasets are clean up and they are not left in the users's cluster.

For example the `coredns-5976949f76` is the one with the patch, `coredns-687db6485c` is the one without the patch.
```
$ kubectl get rs --all-namespaces
NAMESPACE     NAME                 DESIRED   CURRENT   READY     AGE
kube-system   coredns-5976949f76   1         1         1         1m
kube-system   coredns-687db6485c   0         0         0         1m
```

**Comment by ackerman80** [comment]
test-docs-please

**Review by aanm — APPROVED** [review]

File: Documentation/kubernetes/install/eks.rst [hunk 1]
```diff
@@ -54,18 +54,32 @@ Create EKS Cluster
...
+       # if using kube-dns
+       kubectl patch -n kube-system deployment/kube-dns --type merge -p '{"spec":{"template":{"metadata":{"labels":{"io.cilium.fixed-identity":"kube-dns"}}}}}'
+       
+       # if using coredns
+       kubectl patch -n kube-system deployment/coredns --type merge -p '{"spec":{"template":{"metadata":{"labels":{"io.cilium.fixed-identity":"kube-dns"}}}}}'
 
-   This steps allows Cilium to bring up kube-dns networking and enforce
-   security policies before etcd is up.
+   This step allows Cilium to bring up kube-dns networking and enforce
+   security policies before etcd is up. (Note: By default, kubernetes keeps the old replicaset
```
Comment (ianvernon) [inline Documentation/kubernetes/install/eks.rst]: kubernetes --> Kubernetes

replicaset --> ReplicaSet (maybe surround in \`\`)?

File: Documentation/kubernetes/install/eks.rst [hunk 2]
```diff
@@ -54,18 +54,32 @@ Create EKS Cluster
...
+       kubectl patch -n kube-system deployment/kube-dns --type merge -p '{"spec":{"template":{"metadata":{"labels":{"io.cilium.fixed-identity":"kube-dns"}}}}}'
+       
+       # if using coredns
+       kubectl patch -n kube-system deployment/coredns --type merge -p '{"spec":{"template":{"metadata":{"labels":{"io.cilium.fixed-identity":"kube-dns"}}}}}'
 
-   This steps allows Cilium to bring up kube-dns networking and enforce
-   security policies before etcd is up.
+   This step allows Cilium to bring up kube-dns networking and enforce
+   security policies before etcd is up. (Note: By default, kubernetes keeps the old replicaset 
+   but those sets are not running. When the deployment is deleted all replicasets will be cleaned up
```
Comment (ianvernon) [inline Documentation/kubernetes/install/eks.rst]: replicaset --> ReplicaSets

File: Documentation/kubernetes/install/eks.rst [hunk 3]
```diff
@@ -54,18 +54,32 @@ Create EKS Cluster
...
-       kubectl label -n kube-system pod $(kubectl -n kube-system get pods -l k8s-app=kube-dns -o jsonpath='{range .items[]}{.metadata.name}{" "}{end}') io.cilium.fixed-identity=kube-dns
+       # if using kube-dns
+       kubectl patch -n kube-system deployment/kube-dns --type merge -p '{"spec":{"template":{"metadata":{"labels":{"io.cilium.fixed-identity":"kube-dns"}}}}}'
+       
+       # if using coredns
+       kubectl patch -n kube-system deployment/coredns --type merge -p '{"spec":{"template":{"metadata":{"labels":{"io.cilium.fixed-identity":"kube-dns"}}}}}'
 
-   This steps allows Cilium to bring up kube-dns networking and enforce
-   security policies before etcd is up.
+   This step allows Cilium to bring up kube-dns networking and enforce
```
Comment (ianvernon) [inline Documentation/kubernetes/install/eks.rst]: you surround kube-dns with \`\` above, but not here. Do so for consistency.

**Review by ianvernon — DISMISSED** [review]

**Comment by ackerman80** [comment]
test-docs-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**